src/index.ts

```diff
@@ -1,10 +1,16 @@
+import { createIoCContainer } from './ioc/index';
 import { Users } from './services/users';
 import { Logger } from './services/logger';
 
 import type { User, ApiConfig } from './types';
 
+const ioc = createIoCContainer();
+const usersService = ioc.resolve('users');
+const logger = ioc.resolve('logger');
+const http = ioc.resolve('http');
```

Comment:
you don't need to resolve http service here, because you don't use it here directly.